Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2535,16 +2533,32 @@ def visit_DefNode(self, node):
                     error(pxd_def.pos, "previous declaration here")
                 return None
             node = node.as_cfunction(pxd_def)
+        return node
+
+    def visit_ExprNode(self, node):
+        # ignore lambdas and everything else that appears in expressions
+        return node
+
+
+class AutoCpdefFunctionDefinitions(AlignFunctionDefinitions):
+
+    def visit_ModuleNode(self, node):
+        self.directives = node.directives
+        self.imported_names = set()  # hack, see visit_FromImportStatNode()
+        return super(AutoCpdefFunctionDefinitions, self).visit_ModuleNode(node)
+
+    def visit_DefNode(self, node):
+        pxd_def = self.scope.lookup(node.name)
+        if pxd_def and (not pxd_def.scope or not pxd_def.scope.is_builtin_scope):
+            pass
         elif (self.scope.is_module_scope and self.directives['auto_cpdef']
               and node.name not in self.imported_names
               and node.is_cdef_func_compatible()):
-            # FIXME: cpdef-ing should be done in analyse_declarations()
             node = node.as_cfunction(scope=self.scope)
-        # Enable this when nested cdef functions are allowed.
-        # self.visitchildren(node)
         return node
 
     def visit_FromImportStatNode(self, node):
+        # return node
```

Comment:
?

Reply:
This is debug code I have missed to delete.